File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, n: List[int], target: int) -> int:
        left = 0
        right  =  len(n)-1
        while left <= right:
            mid = left  + (right -left) // 2
            
            if target < n[mid]:
                right  = mid -1
            elif target > n[mid]:
                left  = mid +1
            if target == n[mid]:
                return mid

        return -1
        

File: Data Structures & Algorithms/depth-of-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {

    public int maxDepth(TreeNode root) {
        if (root == null) {
            return 0;
        }
        
        // Calculate each side ONLY ONCE
        int leftHeight = maxDepth(root.left);
        int rightHeight = maxDepth(root.right);
        
        // Use Math.max for cleaner logic
        return Math.max(leftHeight, rightHeight) + 1;
    }
    
}
